samples/sample_decode/src/pipeline_decode.cpp

```diff
@@ -345,14 +345,13 @@ mfxStatus CDecodingPipeline::Init(sInputParams *pParams)
         m_memType = pParams->memType;
     else
     {
+#if defined(_WIN32) || defined(_WIN64)
+        m_memType = pParams->bUseHWLib ? D3D11_MEMORY : SYSTEM_MEMORY;
+#elif defined(LIBVA_SUPPORT)
```

Comment:
LGTM but seems you're missing one aspect:
which d3dX type will be used in MSDK when app sends system memory? It should be D3D11. 
Please pay attention to https://github.com/Intel-Media-SDK/MediaSDK/blob/0be32d7f99f7e1eef6c4329f4892e5e0f14128b1/samples/sample_decode/src/pipeline_decode.cpp#L271

Please extend this and add an code comment what we prefer D3D11 by default.